### PR TITLE
feat(ProjectTodo): Remove ProjectTodo and UI changes for Obligation and ProjectTodo 

### PR DESCRIPTION
**Pull request by JaideepPalit**
> - Renamed type to obligationType ,Changed required licenseIds to optional , Added optional ObligationLevel obligationLevel in ObligationStatusInfo
> - Remove ProjectTodo and UI changes for Obligation and ProjectTodo 

> * Which issue is this pull request belonging to and how is it solving it? (*Refer to issue here*)
> * Did you add or update any new dependencies that are required for your change?

> Have you implemented any additional tests? - No

### Checklist
Must:
- [X] All related issues are referenced in commit messages and in PR

Signed-off-by: Jaideep Palit <jaideep.palit@siemens.com>

### CODE REVIEW

File: scripts/migrations/037_checkfor_project_todos_in_moderations.py [hunk 1]
```diff
@@ -0,0 +1,152 @@
...
+            todoslenghtinprojAddn = len(projectAdditions["todos"])
+
+        if todo in projectDeletions:
+            todoslenghtinprojDeln = len(projectDeletions["todos"])
+
+        if (todoslenghtinprojAddn is not None and todoslenghtinprojAddn > 0) or (todoslenghtinprojDeln is not None and todoslenghtinprojDeln > 0):
+            moderationList.append(moderation.get("_id"))
+
+    if len(moderationList) > 0:
+        for x in moderationList:
```
Comment (JaideepPalit) [inline scripts/migrations/037_checkfor_project_todos_in_moderations.py:114]: It would be good to have valid variable name in place of x

**Review by JaideepPalit — COMMENTED** [review]
Kindly look at some minor review comment

File: scripts/migrations/039_projecttodo_to_obligationlist.py [hunk 2]
```diff
@@ -0,0 +1,188 @@
...
+                    todotext = getOblTextNLevel(todo.get("todoId")).get("oblText")
+                    obligationLevel = getOblTextNLevel(todo.get("todoId")).get("oblLevel")
+                    obligationStatusInfo["obligationLevel"] = obligationLevel
+                    obligationStatusInfo["text"] = todotext
+                    obligationStatusInfo["status"] = 'FULFILLED' if todo.get("fulfilled") == True else 'OPEN'
+                    todocomment = todo.get("comments")
+                    obligationStatusInfo["comment"] = todocomment if todocomment is not None else ""
+                    obligationStatusInfo["modifiedOn"] = getDateOnly(todo.get("updated")) if todo.get("updated") != "" else ""
+                    obligationStatusInfo["modifiedBy"] = getEmailFromUserId(todo.get("userId"))
+                    obligationStatusInfo["id"] = todo.get("todoId")
```
Comment (JaideepPalit) [inline scripts/migrations/039_projecttodo_to_obligationlist.py:110]: (from discussion) - Id in obligationStatusInfo is for external id and not couch db doc id

File: backend/src/src-licenseinfo/src/main/java/org/eclipse/sw360/licenseinfo/outputGenerators/DocxGenerator.java [hunk 3]
```diff
@@ -577,7 +576,7 @@ private void printSecondTableOnwards(XWPFDocument document, XmlCursor cursor, Ma
             tableRow.getCell(4);
 
         }
-
+/*
```
Comment (smrutis1) [inline backend/src/src-licenseinfo/src/main/java/org/eclipse/sw360/licenseinfo/outputGenerators/DocxGenerator.java:579]: Please consider to remove commented methods from this file

File: frontend/sw360-portlet/src/main/java/org/eclipse/sw360/portal/portlets/projects/ProjectPortlet.java [hunk 4]
```diff
@@ -238,6 +241,20 @@ public void serveResource(ResourceRequest request, ResourceResponse response) th
             JSONObject dataForChangeLogs = changeLogsPortletUtilsPortletUtils.serveResourceForChangeLogs(request,
                     response, action);
             writeJSON(request, response, dataForChangeLogs);
+        } else if (PortalConstants.LOAD_OBLIGATIONS_VIEW.equals(action)
```
Comment (smrutis1) [inline frontend/sw360-portlet/src/main/java/org/eclipse/sw360/portal/portlets/projects/ProjectPortlet.java:244]: Only `request.setAttribute("inProjectDetailsContext", xxxx);` statement seems to only difference in conditions

File: libraries/lib-datahandler/src/main/java/org/eclipse/sw360/datahandler/common/SW360Utils.java [hunk 5]
```diff
@@ -699,4 +588,95 @@ public static Integer parseStringToNumber(String input) {
             return 0;
         }
     }
+
+    public static Map<String, ObligationStatusInfo> getProjectObligationToDisplay(
```
Comment (smrutis1) [inline libraries/lib-datahandler/src/main/java/org/eclipse/sw360/datahandler/common/SW360Utils.java:592]: `getProjectObligationToDisplay(-,-)`, `getOrganisationObligationToDisplay(-,-)`, `getComponentObligationToDisplay(-,-)`  - These methods contains mostly duplicated codes. Would be good if can be moved to a single method

**Review by smrutis1 — CHANGES_REQUESTED** [review]

File: scripts/migrations/039_projecttodo_to_obligationlist.py (hunk 2 above)
Comment (smrutis1) [reply scripts/migrations/039_projecttodo_to_obligationlist.py]: removed it

File: scripts/migrations/037_checkfor_project_todos_in_moderations.py (hunk 1 above)
Comment (smrutis1) [reply scripts/migrations/037_checkfor_project_todos_in_moderations.py]: refactored

**Comment by JaideepPalit** [comment]
Thanks for review. Made changes

**Review by smrutis1 — APPROVED** [review]

**Comment by JaideepPalit** [comment]
Migration script looks fine to me

**Pull request merged**